addons/mail/static/src/core/common/composer.js

```diff
@@ -210,6 +210,9 @@ export class Composer extends Component {
         if (this.props.placeholder) {
             return this.props.placeholder;
         }
+        if (this.thread?.read_only) {
+            return _t("You don't have the permission to post messages in this channel");
+        }
```

Comment:
Composer placeholder is narrow and can overflow, especially in chat window:
<img width="364" alt="Screenshot 2024-10-21 at 15 28 01" src="https://github.com/user-attachments/assets/99feb437-239f-4443-b0df-4fbaf8199d4b">

I'd suggest to play with textarea value, disabling the textarea, text shorter (easier to read and inline at least in en_US), and style in a way that looks like a disabled composer:
<img width="370" alt="Screenshot 2024-10-21 at 15 30 41" src="https://github.com/user-attachments/assets/a01e2fe7-36a2-4f7d-b6e6-c92fe96bedb8">
(composer actions have been removed too, because no reason to show them)

Reply:
Actions were supposed to be hidden, but I guess I made a rebase mistake.

Reply:
Sorry for confusion, I haven't tested the PR functionally, just saw that composer was using placeholder and I crafted locally the problem with placeholder overflow to show my point.

If we make label more brief (channel is read-only) then this becomes a non-issue for the most part

Reply:
I have updated the placeholder